### DISCUSSION TITLE
Bulk adding jobs to a batch - how to combine bulk and batch?

### DISCUSSION THREAD

**Discussion by amo13** [bensheldon/good_job#1186] [Q&A] [answered]
I have a potentially large amount of jobs that I want to enqueue and run in a batch. I also read about the bulk enqueuing functionality and thought, I'd combine them:
```ruby
update_batch = GoodJob::Batch.new(on_finish: MyCallbackJob) do
  GoodJob::Bulk.enqueue do
    [lots_of_things_needing_update].each do |thing|
      MyUpdateJob.perform_later(thing: thing, target_version: "1.2.3")
    end
  end
end
update_batch.description = "Update all things to 1.2.3"
update_batch.save
update_batch.enqueue
```

But somehow, the batch ends up being empty. How do I combine those to enqueue all jobs in a batch and also optimize the query to the database?

**Comment by bensheldon** [comment, marked as answer]
A Batch already Bulk enqueues.

`GoodJob::Batch` uses it's own bulking wrapper (a `GoodJob::Bulk::Buffer`, which is the lower-level object also used by `GoodJob::Bulk`) to capture jobs, so creating a new `GoodJob::Bulk` will intercept those jobs and not add them to the batch:

https://github.com/bensheldon/good_job/blob/ae7448254729520a83fedf2d23e75add3500b0e6/app/models/good_job/batch.rb#L119-L131
